velox/dwio/common/tests/MemorySinkTest.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/common/FileSink.h"
+
+#include <gtest/gtest.h>
+
+namespace facebook::velox::dwio::common {
+
+class MemorySinkTest : public testing::Test {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+  }
+  std::shared_ptr<velox::memory::MemoryPool> pool_{
+      memory::memoryManager()->addLeafPool()};
+};
+
+TEST_F(MemorySinkTest, create) {
+  std::string chars("abcdefghijklmnopqrst");
+  std::vector<DataBuffer<char>> buffers;
+
+  buffers.emplace_back(*pool_);
+  for (auto i = 0; i < 10; ++i) {
+    buffers.back().append(chars[i]);
+  }
+
+  buffers.emplace_back(*pool_);
+  for (auto i = 10; i < chars.length(); ++i) {
+    buffers.back().append(chars[i]);
+  }
+
+  EXPECT_EQ(buffers.size(), 2);
```

Comment:
Perhaps we do not need this check, and it should be better to use a for loop to populate the buffers.